cmd/argocd/commands/app.go

```diff
@@ -705,7 +705,11 @@ func NewApplicationDiffCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 						errors.CheckError(err)
 						key = kube.GetResourceKey(target)
 					}
-
+					if key.Kind == "Secret" {
```

Comment:
Please add group check as well to make sure it is not some CRD with Secret kind. Also we have [constants](https://github.com/argoproj/argo-cd/blob/master/util/kube/kube.go#L40) for popular kinds. Can you please use it:


```
if key.Kind == kube.SecretKind && key.Group == "" {
```

Reply:
LGTM. merging it and will add group check in next PR along with test

Reply:
sorry didn't have time to work on this earlier, thanks for cleaning up!